### PR TITLE
Add auto complete to tags

### PR DESCRIPTION
**Pull request by verdefred**
## Summary
Show total of users with each tag and add auto complete to tags dropdown on the Edit Profile

## Notion link

https://www.notion.so/talentprotocol/Tags-Show-totals-and-add-auto-complete-b7fe33e283a441369574cbe1fee87bd3

### CODE REVIEW

File: app/controllers/api/v1/tags_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class API::V1::TagsController < ApplicationController
+  def index
+    tags =
+      Tag
+        .order(:id)
+        .where(hidden: false)
```
Comment (RubenSousaDinis) [inline app/controllers/api/v1/tags_controller.rb:6]: we have a scope: `visible`

**Review by RubenSousaDinis — COMMENTED** [review]
Left some comments. Great work!

File: app/controllers/api/v1/tags_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+        .where("description ilike ?", "%#{params[:description]}%")
+        .limit(7)
```
[the remark this answers is not in the document]
Comment (verdefred) [reply app/controllers/api/v1/tags_controller.rb:8]: it was something discussed with design. they feel that more that 7 will be too much for a dropdown but I can increase it a bit, I guess

File: app/controllers/api/v1/tags_controller.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class API::V1::TagsController < ApplicationController
+  def index
+    tags =
+      Tag
+        .visible
+        .joins(:user_tags)
```
Comment (RubenSousaDinis) [inline app/controllers/api/v1/tags_controller.rb:6]: should we use left join here? If a tag has no usage and it exists in the system it will not appear.

**Review by RubenSousaDinis — COMMENTED** [review]
Just one small comment. We are ready to go after this one!

**Review by RubenSousaDinis — DISMISSED** [review]
Well done!

File: app/packs/src/components/supporters/edit/About.jsx [hunk 4]
```diff
@@ -178,11 +185,27 @@ const About = ({
     changeSharedState((prev) => ({ ...prev, [attribute]: value }));
   };
 
-  const changeTags = (tags) => {
+  const getTags = (tagName) => {
```
Comment (francisco-leal) [inline app/packs/src/components/supporters/edit/About.jsx:188]: let's add a short debounce here so we don't flood the server.

**Review by RubenSousaDinis — APPROVED** [review]
Well done!

**Pull request merged**